lib/resolv.rb

```diff
@@ -2145,8 +2152,10 @@ def hash # :nodoc:
       end
 
       def self.get_class(type_value, class_value) # :nodoc:
-        return ClassHash[[type_value, class_value]] ||
-               Generic.create(type_value, class_value)
+        cached = const_defined?(:"Type#{type_value}_Class#{class_value}") &&
+                 const_get(:"Type#{type_value}_Class#{class_value}")
+
+        return cached || Generic.create(type_value, class_value)
```

Comment:
```suggestion
        cache = "Type#{type_value}_Class#{class_value}"
        return (const_defined?(cache) && const_get(cache)) ||
               Generic.create(type_value, class_value)
```

May it be better to have a separate method to make the cache name?

Reply:
looks better indeed, added.